c2rust-analyze/src/context.rs

```diff
@@ -357,7 +361,16 @@ impl<'a, 'tcx> AnalysisCtxt<'a, 'tcx> {
     }
 
     pub fn project(&self, lty: LTy<'tcx>, proj: &PlaceElem<'tcx>) -> LTy<'tcx> {
-        util::lty_project(lty, proj)
+        let adt_func = |adtdef: AdtDef, field: Field| {
+            let fielddef = adtdef.all_fields().nth(usize::from(field)).unwrap();
```

Comment:
```suggestion
            let fielddef = &adtdef.variant(0usize.into()).fields[field.index()];
```

If `adt_func` knows it's being called on a `struct` and not an `enum`, then this is clearer and more efficient.

Reply:
Actually, use `.non_enum_variant()` here as @spernsteiner said.  I didn't realize that that was a method.